### DISCUSSION TITLE
No BMP encode support?

### DISCUSSION THREAD

**Discussion by Dunge** [mono/SkiaSharp#2417] [General]
I'm migrating my project from System.Drawing.Common to SkiaSharp for cross platform Linux docker usage. Managed to load all my assets in multiple format, draw on them, return jpg for my web service, everything was going fine. Until I ended up trying to save as a .BMP file and got null.

Found https://github.com/mono/SkiaSharp/issues/320 that basically say it's not supported in skia directly, and it's right, from [this page](https://api.skia.org/classSkImage.html#a10e0290eb37dd3c729778eb522fdc19a) it seems only supported on iOS/Mac.

This is pretty frustrating considering [the documentation of skiasharp Encode()](https://learn.microsoft.com/en-us/dotnet/api/skiasharp.skbitmap.encode?view=skiasharp-2.88) doesn't mention it.

I would have thought it wouldn't be so hard to find a library thay cab handle this most basic format.

**Comment by leeboozer63** [comment]
I also find it bizarre that SkiaSharp requires you to work with the bitmap format, but cannot export (Encode) to that same format.

**Comment by mattleibow** [comment]
SkiaSharp can decode a wide variety of codecs - even BMP, but it is focused on drawing to the screen rather than files. Typically, PNG is smaller and more portable than BMP, so adding BMP was no net advantage.

**Comment by zhzy0077** [comment]
What about at least throwing an exception rather than return null for Encode calls? It would save a lot of debugging time.
